File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc,char *argv[])
{
   if(argc != 2)
   {
       printf("Invalid Command Line Arguments");
       return 1;
   }
   int x = atoi(argv[1]);
   printf("plaintext: ");
   string in = get_string();
   int len = strlen(in);
   char out[len];
  
   for(int i=0; i<len; i++)
   {
       if(in[i] >= 'a' && in[i] <= 'z') 
        out[i] = 'a' + (((in[i]-'a')+x)%26);
       else if(in[i] >= 'A' && in[i] <= 'Z') 
        out[i] = 'A' + (((in[i]-'A')+x)%26);
       else
        out[i] = in[i];
   }
  
    printf("ciphertext: %s\n",out);
}

File: pset6/caesar.py
import cs50
import sys


if(len(sys.argv) != 2):
  print("Invalid Command Line Arguments");
  exit(1);

x = int(sys.argv[1]);
print("plaintext: ",end='');
inp = cs50.get_string();
length = len(inp);
out = "";

for i in range(0,length):
   
       if(inp[i] >= 'a' and inp[i] <= 'z'):
        out += chr(ord('a') + (((ord(inp[i])-ord('a'))+x)%26));
       elif(inp[i] >= 'A' and inp[i] <= 'Z'): 
        out += chr(ord('A') + (((ord(inp[i])-ord('A'))+x)%26));
       else:
        out += inp[i];
   
  
print("ciphertext:",out);
